### ISSUE TITLE
New categories for Hazard indices in discipline 20 [Code Table 4.1 and 4.2 and new Code table 4.253]

### ISSUE THREAD

**Issue by sebvi** [wmo-im/GRIB2#285]
### Initial request

Following the discussion in issue #254 , we would like to propose 2 new categories in Code Table 4.2 discipline 20: "Meteorological and hydrological hazard indices"  and "Environmental hazard indices".

The indices in these 2 categories utilize an auxiliary Code Table 4.253. This table is left empty and managed by each centre as we felt it difficult to propose a unique coding of the impact and the severity of the hazards. Our original proposal in #254 contains an example of such table.

### Amendment details

ADD new Categories in Code Table 4.1 discipline 20
 number| Description
-- | --
4 | Meteorological and hydrological hazard indices 
5 | Environmental hazard indices 

ADD new entries in Code table 4.2 discipline 20 category 4
Parameter number | name | Unit
-- | -- | --
0 | Reserved | -
1 | Downburst | [Code table 4.253]
2 | Lightning (Electrical Storm) | [Code table 4.253]
3 | Thunderstorm | [Code table 4.253]
4 | Coastal Flood | [Code table 4.253]
5 | Estuarine (Coastal) Flood | [Code table 4.253]
6 | Flash Flood | [Code table 4.253]
7 | Fluvial (Riverine) Flood | [Code table 4.253]
8 | Groundwater Flood | [Code table 4.253]
9 | Ice-Jam Flood Including Debris | [Code table 4.253]
10 | Ponding (Drainage) Flood | [Code table 4.253]
11 | Snowmelt Flood | [Code table 4.253]
12 | Surface Water Flooding | [Code table 4.253]
13 | Glacial Lake Outburst Flood | [Code table 4.253]
14 | Black Carbon (Brown Clouds) | [Code table 4.253]
15 | Dust storm or Sandstorm | [Code table 4.253]
16 | Fog | [Code table 4.253]
17 | Haze | [Code table 4.253]
18 | Polluted Air | [Code table 4.253]
19 | Sand haze | [Code table 4.253]
20 | Smoke | [Code table 4.253]
21 | Ocean Acidification | [Code table 4.253]
22 | Rogue Wave | [Code table 4.253]
23 | Sea Water Intrusion | [Code table 4.253]
24 | Sea Ice (Ice Bergs) | [Code table 4.253]
25 | Ice Flow | [Code table 4.253]
26 | Seiche | [Code table 4.253]
27 | Storm Surge | [Code table 4.253]
28 | Storm Tides | [Code table 4.253]
29 | Tsunami | [Code table 4.253]
30 | Depression or Cyclone (Low Pressure Area) (Pressure-Related) | [Code table 4.253]
31 | Extra-tropical Cyclone (Pressure-Related) | [Code table 4.253]
32 | Sub-Tropical Cyclone (Pressure-Related) | [Code table 4.253]
33 | Acid Rain | [Code table 4.253]
34 | Blizzard | [Code table 4.253]
35 | Drought | [Code table 4.253]
36 | Hail | [Code table 4.253]
37 | Ice Storm | [Code table 4.253]
38 | Snow | [Code table 4.253]
39 | Snow Storm | [Code table 4.253]
40 | Cold Wave | [Code table 4.253]
41 | Dzud | [Code table 4.253]
42 | Freeze | [Code table 4.253]
43 | Frost (Hoar Frost) | [Code table 4.253]
44 | Freezing Rain (Supercooled Rain) | [Code table 4.253]
45 | Glaze | [Code table 4.253]
46 | Ground Frost | [Code table 4.253]
47 | Heatwave | [Code table 4.253]
48 | Icing (Including Ice) | [Code table 4.253]
49 | Thaw | [Code table 4.253]
50 | Avalanche | [Code table 4.253]
51 | Mud Flow | [Code table 4.253]
52 | Rock slide | [Code table 4.253]
53 | Derecho | [Code table 4.253]
54 | Gale (Strong Gale) | [Code table 4.253]
55 | Squall | [Code table 4.253]
56 | Subtropical Storm | [Code table 4.253]
57 | Tropical Cyclone (Cyclonic Wind, Rain [Storm] Surge) (Wind-Related) | [Code table 4.253]
58 | Tropical Storm (Wind-Related) | [Code table 4.253]
59 | Tornado | [Code table 4.253]
60 | Wind | [Code table 4.253]
191 - 254 | Reserved for local use | -
255 | Missing | -

ADD new entries in Code table 4.2 discipline 20 category 5
Parameter number | name | Unit
-- | -- | --
0 | Reserved | -
1 | Household Air Pollution | [Code table 4.253]
2 | Air Pollution (Point Source) | [Code table 4.253]
3 | Ambient (Outdoor) Air Pollution | [Code table 4.253]
4 | Land Degradation | [Code table 4.253]
5 | Soil Degradation | [Code table 4.253]
6 | Runoff / Nonpoint Source Pollution | [Code table 4.253]
7 | Salinity | [Code table 4.253]
8 | Biodiversity Loss | [Code table 4.253]
9 | Deforestation | [Code table 4.253]
10 | Forest Declines and Diebacks | [Code table 4.253]
11 | Forest Disturbances | [Code table 4.253]
12 | Forest Invasive Species | [Code table 4.253]
13 | Wildfires | [Code table 4.253]
14 | Desertification | [Code table 4.253]
15 | Loss of Mangroves | [Code table 4.253]
16 | Wetland Loss/Degradation | [Code table 4.253]
17 | Coral Bleaching | [Code table 4.253]
18 | Compressive Soils | [Code table 4.253]
19 | Soil Erosion | [Code table 4.253]
20 | Coastal Erosion and Shoreline Change | [Code table 4.253]
21 | Permafrost Loss | [Code table 4.253]
22 | Sand Mining | [Code table 4.253]
23 | Sea Level Rise | [Code table 4.253]
24 | Eutrophication | [Code table 4.253]
191 - 254 | Reserved for local use | -
255 | Missing | -

ADD new Code table 4.253 Hazard Index
Code | Description
-- | --
0 | Undefined
1-191 | Reserved (see Note)
192-254 | Reserved for local use
255 | Missing

</div>
<p>Note: The hazard index can be locally defined by the centre</p>

### Requestor(s)

Sebastien Villaume (ECMWF)
Matthew  Griffith (ECMWF)

### Stakeholder(s)

ECMWF, Destination Earth service

### Publication(s)

Manual on Codes (WMO-No. 306), Volume I.2, GRIB code table 4.2, discipline 20, category 4 (create)
Manual on Codes (WMO-No. 306), Volume I.2, GRIB code table 4.2, discipline 20, category 5 (create)
Manual on Codes (WMO-No. 306), Volume I.2, GRIB code table 4.253 (create)

### Expected impact of change

None

### References
UN Office for disaster risk reduction which classifies the various type of hazards: [UNDRR document](https://www.undrr.org/media/73913/download?startDownload=20240426)

**Comment by amilan17** [comment]
https://github.com/wmo-im/tt-tdcf/wiki/2024.11.13.tt.tdcf notes: 
- from UNDRR  publication
- @sebvi update table 4.253 -- no need to have missing, reserved, etc.

**Comment by sebvi** [comment]
Please find attached some sample files and a version of ecCodes capable of encode/decode the new proposed templates.
[eccodes-2.40.0-Source.tar.gz](https://github.com/user-attachments/files/17976054/eccodes-2.40.0-Source.tar.gz)
[FT2025-1_285.tar.gz](https://github.com/user-attachments/files/17976057/FT2025-1_285.tar.gz)

**Comment by amilan17** [comment]
https://github.com/wmo-im/et-data/wiki/2024.12.03.et.data notes:
no additional encoding/decoding testing needed because they are just additional parameters
@sebvi branch is ready for updating

**Comment by sebvi** [comment]
branch is updated

**Comment by amilan17** [comment]
https://github.com/wmo-im/et-data/wiki/2025.01.14.et.data notes: 
check branch, not validation necessary

**Comment by amilan17** [comment]
> * update table 4.253 -- no need to have missing, reserved, etc.

@sebvi I think the team recommended some changes to code table 4.253 in November '24

Can you update the parameter names to begin with caps and then lower case for the rest?  
e,g., Lightning (Electrical Storm) --> Lightning (electrical storm)

**Comment by sebvi** [comment]
yes, it is correct. Since the values from this table are encoded in the data section we could make it even bigger
